test/lsptoolshost/integrationTests/formatting.integration.test.ts

```diff
@@ -91,4 +99,60 @@ describe(`Formatting Tests`, () => {
         ];
         await expectText(vscode.window.activeTextEditor!.document, expectedText);
     });
+
+    test('Document formatting can organize imports', async () => {
+        await setOrganizeImportsOnFormat(true);
+        await activateCSharpExtension();
+
+        await formatDocumentAsync();
+
+        const expectedText = [
+            'using System;',
+            'using Options;',
+            'namespace Formatting;',
+            'class DocumentFormatting',
+            '{',
+            '    public int Property1',
+            '    {',
+            '        get; set;',
+            '    }',
+            '',
+            '    public void Method1()',
+            '    {',
+            '        System.Console.Write("");',
+            '    }',
+            '}',
+        ];
+        await expectText(vscode.window.activeTextEditor!.document, expectedText);
+    });
+
+    test('Document range formatting does not organize imports', async () => {
+        await setOrganizeImportsOnFormat(true);
+        await activateCSharpExtension();
+
+        await formatRangeAsync(new vscode.Range(0, 0, 7, 0));
+
+        const expectedText = [
+            'using Options;',
+            'using System;',
+            'namespace Formatting;',
+            'class DocumentFormatting',
+            '{',
+            '    public int Property1',
+            '    {',
+            '        get; set;',
+            '    }',
+            '',
+            '    public void Method1() {',
+            '            System.Console.Write("");',
+            '    }',
+            '}',
+        ];
+        await expectText(vscode.window.activeTextEditor!.document, expectedText);
+    });
+
+    async function setOrganizeImportsOnFormat(value: boolean | undefined) {
+        const dotnetConfig = vscode.workspace.getConfiguration('dotnet');
+        await dotnetConfig.update('formatting.organizeImportsOnFormat', value, true);
```

Comment:
I suspect we might hit test flakiness here - unfortunately we don't currently have a way to know if the server actually updated the configuration (option updates are sent as notifications).  

We may want to restart the server after setting this option.